File: 2 (7.1)/1978/1978.cs
// 주어진 수 N개 중에서 소수가 몇 개인지 찾아서 출력
// input: 4 1 3 5 7, output: 3

using System;

namespace _1978 {
    class Program {
        public static void Main(String[] args) {
            int N = int.Parse(Console.ReadLine());
            String[] tokens = Console.ReadLine().Split(' ');
            int count = 0;

            for (int i = 0; i < N; i++) {
                int num = int.Parse(tokens[i]);
                if (IsPrime(num)) count++; // 소수이면 count 증가
            }

            Console.WriteLine(count);
        }

        public static bool IsPrime(int num) {
            if (num < 2) return false;          // 2미만은 소수가 아님
            for (int i = 2; i * i <= num; i++) {
                if (num % i == 0) return false; // 소수가 아님
            }
            return true; // 소수
        }
    }
}

File: 2 (7.1)/1978/1978.java
// 주어진 수 N개 중에서 소수가 몇 개인지 찾아서 출력
// input: 4 1 3 5 7, output: 3

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        // 공백(" ")으로 구분된 N개의 수를 입력받아서 배열에 저장
        int[] arr = new int[N];
        for (int i = 0; i < N; i++) {
            arr[i] = sc.nextInt();
        }
        int count = 0;

        for (int i = 0; i < N; i++) {
            if (isPrime(arr[i])) count++;
        }

        System.out.println(count);
    }

    public static boolean isPrime(int n) {
        if (n < 2) return false;
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) return false;
        }
        return true;
    }
}